main/src/ca/uwaterloo/flix/language/phase/unification/BoolMinimization.scala

```diff
@@ -40,9 +39,13 @@ object BoolMinimization {
 
   }
 
+  /**
+    * Sort terms `True` -> `False` -> `Var`s (by name then negation) ->
+    * complex negation -> conjunctions -> disjunctions.
+    */
   def sortTerms(terms: List[Formula]): List[Formula] = {
     import Formula._
-    terms.sortWith{
+    terms.sortWith {
       case (True, True) => false
       case (True, False | Var(_) | Not(_) | And(_) | Or(_)) => true
```

Comment:
This seems to break reflectivity. Maybe instead of sorting we could just filter and then create a set of the variables?

Reply:
> This seems to break reflectivity

How so? The function required is supposed to be `<` to have a stable sort, so for equal things they should return false.

> Maybe instead of sorting we could just filter and then create a set of the variables?

So have `And(terms: List[Formula], vars: Set[Var])` or something like that?

Reply:
An order should be reflexive, i.e.  x <= x. 

I would keep the representation the same. 

What you could do is given a CNF you can find all the variable clauses.

Reply:
Maybe you do not ever care about sorting. You just want unit propagation.

Reply:
> An order should be reflexive, i.e. x <= x.

But this is `<`, so `x < x == false` is expected

> Maybe you do not ever care about sorting. You just want unit propagation.

I used sorting for two reasons.
(1) it made cnf slightly easier the way I coded it (which isn't very polished)
(2) it makes it easy to check for `x` and `!x` since they are next to each other

Reply:
To sort requires a total order and a total order must be reflexive. So I am not sure exactly what you mean. Its possible it works accidentally because sorthWith uses eq/== internally, but I still think the implementation is not what you want.

Reply:
I would propose to forget about sort, but instead write something like:

```
def unitClauses(): List[Formula]
```

and then just use that. 

Does sorting really buy us anything?

Reply:
(I mean I thought it did in the beginning, but now I don't see why its useful).

Reply:
> To sort requires a total order and a total order must be reflexive. So I am not sure exactly what you mean. Its possible it works accidentally because sorthWith uses eq/== internally, but I still think the implementation is not what you want.

After googling this is called a strict total order but it can describe the same ordering. Eq can be checked by `!(x < y) && !(y < x)`. Instead of being reflective this ordering is anti symmetric so its not just a partial ordering

Reply:
Okay- but I guess in any case we will not need to sort

Reply:
> Does sorting really buy us anything?

I'm not sure no. We could also just have `And(vars: List[Var], not: List[Not], conj: List[And], disj: List[Or])` or something like that. Then we will never need the linear pass to collect the unit terms or to order them (except when looking for x and not x)